v4/source/actions.yaml

```diff
@@ -0,0 +1,116 @@
+  '/actions/dialogs/open':
+    post:
+      tags:
+        - integration_actions
+      summary: Open a dialog
+      description: |
+        Open an interactive dialog using a trigger ID provided by a slash command, or some other action payload. See https://docs.mattermost.com/developer/interactive-dialogs.html for more information on interactive dialogs.
+        __Minimum server version: 5.6__
+      parameters:
+        - name: body
+          in: body
+          description: Metadata for the dialog to be opened
+          required: true
+          schema:
+            type: object
+            required:
+              - trigger_id
+              - url
+              - dialog
+            properties:
+              trigger_id:
+                type: string
+                description: Trigger ID provided by other action
+              url:
+                type: object
+                description: The URL to send the submitted dialog payload to
+              dialog:
+                type: object
+                required:
+                  - title
+                  - elements
+                description: Post object to create
+                properties:
+                  callback_id:
+                    type: string
+                    description: Set an ID that will be included when the dialog is submitted
+                  title:
+                    type: string
+                    description: Title of the dialog
+                  elements:
+                    type: array
+                    description: Input elements, see https://docs.mattermost.com/developer/interactive-dialogs.html#elements
+                    items:
+                      type: object
+                  submit_label:
+                    type: string
+                    description: Label on the submit button
+                  notify_on_cancel:
+                    type: boolean
+                    description: Set true to receive payloads when user cancels a dialog
+                  state:
+                    type: string
+                    description: Set some state to be echoed back with the dialog submission
+      responses:
+        '200':
+          description: Dialog open successful
+          schema:
+            $ref: '#/definitions/StatusOK'
+        '400':
+          $ref: '#/responses/BadRequest'
+
+  '/actions/dialogs/submit':
+    post:
+      tags:
+        - integration_actions
+      summary: Submit a dialog
+      description: |
+        Endpoint used by the Mattermost clients to submit a dialog. See https://docs.mattermost.com/developer/interactive-dialogs.html for more information on interactive dialogs.
+        __Minimum server version: 5.6__
+      parameters:
+        - name: body
+          in: body
+          description: Dialog submission data
+          required: true
+          schema:
+            type: object
+            required:
+              - url
```

Comment:
Is `url` required? I asked since SubmitDialogRequest says it's "omitempty".

Reply:
It's required, the omitempty is for when it's used in a different context